### PR TITLE
fix: treat exceptions from jsonwebtoken

### PR DESCRIPTION
**Pull request by r4mmer**

### CODE REVIEW

File: src/api/auth.ts [hunk 1]
```diff
@@ -180,10 +180,25 @@ export const bearerAuthorizer: APIGatewayTokenAuthorizerHandler = async (event)
...
+    data = jwt.verify(
+      sanitizedToken,
+      process.env.AUTH_SECRET,
+    );
+  } catch (e) {
+    // XXX: find a way to return specific error to frontend or make all errors Unauthorized?
+    //
+    // Identify exception from jsonwebtoken by the name property
+    // https://github.com/auth0/node-jsonwebtoken/blob/master/lib/TokenExpiredError.js#L5
+    if (e.name === 'JsonWebTokenError') {
```
Comment (andreabadesso) [inline src/api/auth.ts:194]: What is this error? An invalid JWT was sent from the user?

File: src/api/auth.ts [hunk 2]
```diff
@@ -180,10 +180,25 @@ export const bearerAuthorizer: APIGatewayTokenAuthorizerHandler = async (event)
...
-    sanitizedToken,
-    process.env.AUTH_SECRET,
-  );
+  let data;
+  try {
+    data = jwt.verify(
+      sanitizedToken,
+      process.env.AUTH_SECRET,
+    );
+  } catch (e) {
```
Comment (andreabadesso) [inline src/api/auth.ts:189]: I think we should log every unhandled error here

File: src/api/auth.ts [hunk 3]
```diff
@@ -180,10 +180,25 @@ export const bearerAuthorizer: APIGatewayTokenAuthorizerHandler = async (event)
...
+    // XXX: find a way to return specific error to frontend or make all errors Unauthorized?
```
Comment (andreabadesso) [inline src/api/auth.ts:190]: I think it would be nice to inform the user the type of error, if possible

**Review by andreabadesso — CHANGES_REQUESTED** [review]

File: src/api/auth.ts (hunk 1 above)
Comment (r4mmer) [reply src/api/auth.ts:194]: This is the most generic error, invalid jwt (like `"Bearer 123"`) or secret mismatch, any error that prevents the lib from verifying that this token is valid except for errors related to the headers `exp` and `nbf`.

File: src/api/auth.ts (hunk 2 above)
Comment (r4mmer) [reply src/api/auth.ts:189]: Done.

File: src/api/auth.ts (hunk 3 above)
Comment (r4mmer) [reply src/api/auth.ts:190]: I need to test this, but if the error thrown is an instance of `Error`, the API Gateway will return the message inside to the user along with a 401 (i think). I don't like exposing internal errors to the user but we could do it.

**Review by pedroferreira1 — APPROVED** [review]

File: src/api/auth.ts (hunk 3 above)
Comment (andreabadesso) [reply src/api/auth.ts:190]: I agree with not exposing internal errors, but we are in control of the error that will get thrown here, right?

We could just throw an error with the nessage we want inside the catch block

**Review by andreabadesso — APPROVED** [review]

**Pull request merged**